amundsen_application/static/js/components/common/SearchBar/index.tsx

```diff
@@ -88,23 +90,16 @@ export class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
 
   handleValueSubmit = (event: React.FormEvent<HTMLFormElement>) : void => {
     const searchTerm = this.state.searchTerm.trim();
+    const useFilters = this.props.location ? this.props.location.pathname === '/search' : false;
```

Comment:
Is there a better place for this logic? If possible I wouldn't want the `SearchBar` to be aware of `location` and which routes should preserve the filters. 

- Maybe the reducer or saga should be aware of the route and decide if the filters should be used or not.
- Maybe there should be a small checkbox somewhere that would explicitly decide to preserve or clear filters.

Reply:
Hope the following is clear I time-boxed some investigation:

1. After some investigation I'm not confident this is a quick fix as opposed to separate task of augmenting our usage of Redux on that manner. I couldn't quite wrap my head around the safest way to do this that is 100% guaranteed to keep the redux state and react-router in sync -- if it exists.

2. If the ability to toggle `useFilters` or not was equal work effort I'd be confident to include it, but in this case there would be extra effort to add this to the application state and ensure page navigations still somehow update this state. I think it's worth investing effort into that toggle implementation only if users request it, or Design supports it. I'd essentially hate to spend a couple days on this and then have to remove it after Design QA. It is worth proposing as part of the final design QA though.

In summary we could talk about this more offline as its own task. I boil this down to the fact that we need a `SearchBar` that will cause either one of two action chains to happen:
1. clearFilters & submitSearch 
2. submitSearch

Meaning that `SearchBar` could take its submit method as a passed property, and it's parent components which are connected to the router could pass the correct submit action to the `SearchBar`. Let me know what you think, and I can add the modification before final feature branch review.

Reply:
FYI references -- first found [this library](https://github.com/reactjs/react-router-redux) which might have been overcomplicating things and is now deprecated, but it could also be as little effort as some of[ these suggestions](https://www.pluralsight.com/guides/firing-redux-actions-on-route-transitions)